### PR TITLE
Darwin CohortDiagnostics release

### PR DESCRIPTION
**Pull request by ablack3**
@rossdwilliams @ginberg 

This is the first Darwin release of CohortDiagnostics fork. We will try to change the package in small steps working toward full implementation of the Darwin SRS.

@ginberg will maintain the fork and be responsible for merging PRs to main and creating release tags.

### CODE REVIEW

**Comment by ginberg** [comment]
@ablack3 the R check is failing, can you have a look at that?

File: R/ConceptIds.R [hunk 1]
```diff
@@ -44,19 +28,21 @@ exportConceptInformation <- function(connection = NULL,
...
+    tolower(getTableNames(connection, cdmDatabaseSchema))
   vocabularyTablesInCdmDatabaseSchema <-
     tablesInCdmDatabaseSchema[tablesInCdmDatabaseSchema %in% vocabularyTableNames]
 
   if (length(vocabularyTablesInCdmDatabaseSchema) == 0) {
     stop("Vocabulary tables not found in ", cdmDatabaseSchema)
   }
-  sql <- "SELECT DISTINCT concept_id FROM @unique_concept_id_table;"
+
+  sql <- "SELECT DISTINCT concept_id FROM main.@unique_concept_id_table;"
```
Comment (ginberg) [inline R/ConceptIds.R:43]: we should not hardcode the schema name here (and in other places)

File: R/ConceptSets.R [hunk 2]
```diff
@@ -114,7 +114,7 @@ getParentCohort <- function(cohort, cohortDefinitionSet) {
   ))
 }
 
-combineConceptSetsFromCohorts <- function(cohorts) {
+combineConceptSetsFromCohorts <- function(cohorts) { #browser()
```
Comment (ginberg) [inline R/ConceptSets.R:117]: please remove "#browser"

File: R/RunDiagnostics.R [hunk 3]
```diff
@@ -605,7 +606,7 @@ executeDiagnostics <- function(cohortDefinitionSet,
     }
   )
 
-  cohortDefinitionSet$checksum <- computeChecksum(cohortDefinitionSet$sql)
```
Comment (ginberg) [inline R/RunDiagnostics.R:608]: why is it needed to change this?

File: R/RunDiagnostics.R [hunk 4]
```diff
@@ -724,13 +731,13 @@ executeDiagnostics <- function(cohortDefinitionSet,
     }
   } else {
     if (conceptCountsTable == "#concept_counts") {
-      stop("Temporary conceptCountsTable name. Please provide a valid external ConceptCountsTable name")
+      # stop("Temporary conceptCountsTable name. Please provide a valid external ConceptCountsTable name")
```
Comment (ginberg) [inline R/RunDiagnostics.R:734]: why changing this?

File: R/RunDiagnosticsCdm.R [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+#' @param runAnalysis A numeric vector corresponding to the diagnostic analyses should be executed. See Details
+#' @param minCellCount  The minimum cell count for fields contains person counts or fractions.
+#' 
+#' @details runAnalysis argument
+#' 
+#' Pass a numeric vector matching the following analyses. Negative numbers will be interpreted as exclusions.
+#' Use a numeric vector to select the analyses you want to run. Negative numbers will exclude analyses.
+#' The selection works the same way as vector subsetting in R.
+#' 
+#' # Analyses to choose from:
```
Comment (ginberg) [inline R/RunDiagnosticsCdm.R:35]: I am not against having a runAnalysis argument, but please fill in the descriptions of each analysis

File: R/RunDiagnosticsCdm.R [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+      json = as.character(json),
+      cohortId = as.numeric(cohort_definition_id))
+  
+  executeDiagnostics(cohortDefinitionSet,
+                     connectionDetails = NULL,
+                     connection = attr(cdm, "dbcon"),
+                     cdmVersion = floor(as.numeric(CDMConnector::version(cdm))),
+                     cohortTable = cohortTable,
+                     conceptCountsTable = conceptCountsTable,
+                     # cohortTable = paste(attr(cdm, "write_schema"), cohortTable, sep = "."),
```
Comment (ginberg) [inline R/RunDiagnosticsCdm.R:121]: please remove the commented out lines

File: R/Utils.R [hunk 7]
```diff
@@ -0,0 +1,55 @@
+loadRenderTranslateSql <- function(
```
Comment (ginberg) [inline R/Utils.R:1]: can we have this function in the file "DatabaseFunctions.R"?

File: tests/testthat/test-CDMConnector.R [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  if(!dir.exists(exportFolder)) dir.create(exportFolder)
+  
+  cohortTable <- "mycohort"
+  con <- DBI::dbConnect(duckdb::duckdb(CDMConnector::eunomia_dir()))
+  
+  cdm <- CDMConnector::cdmFromCon(con, cdmSchema = "main", writeSchema = "main", cdmName = "eunomia")
+  
+  cohortSet <- CDMConnector::readCohortSet(system.file("cohorts1", package = "CDMConnector"))
+  
+  # debugonce(executeDiagnosticsCdm)
```
Comment (ginberg) [inline tests/testthat/test-CDMConnector.R:14]: please remove line

File: R/RunDiagnostics.R [hunk 9]
```diff
@@ -238,25 +238,7 @@ executeDiagnostics <- function(cohortDefinitionSet,
                                seed = 64374,
                                seedArgs = NULL,
                                sampleIdentifierExpression = "cohortId * 1000 + seed") {
-  # collect arguments that were passed to cohort diagnostics at initiation
-  callingArgs <- formals(executeDiagnostics)
-  callingArgsJson <-
```
Comment (ginberg) [inline R/RunDiagnostics.R:243]: let's just leave this in, since it's not related to the goal of this PR

File: R/RunDiagnostics.R (hunk 3 above)
Comment (ablack3) [reply R/RunDiagnostics.R:608]: debugging.  changed it back.

File: R/RunDiagnostics.R (hunk 4 above)
Comment (ablack3) [reply R/RunDiagnostics.R:663]: debugging. removed change

File: R/RunDiagnostics.R (hunk 9 above)
Comment (ablack3) [reply R/RunDiagnostics.R:243]: agreed.

**Review by ginberg — APPROVED** [review]

**Pull request merged**